isa.c

```diff
@@ -296,4 +312,85 @@ static void exec_B(uint32_t instruction)
     next_state.regs[PC] = curr_state.regs[PC] + (sign_extend(signed_immed_24,30,24) << 2);
 }
 
+static void exec_LDRB(uint32_t instruction)
+{
+    if (condition_check(curr_state, get_bits(instruction, 31, 28))) {
+        uint8_t data = mem_read_8(ld_str_addr_mode(curr_state, &(next_state), instruction));
+        uint32_t rd_id = get_bits(instruction, 15, 12);
+        next_state.regs[rd_id] = data; // casting uint8_t to uint32_t zeros top 3 bytes on its own; done to store byte to LSB of rd_id
+    }
+}
+
+// Multiply and Multiply-Accumulate Instructions
+// MUL and MLA produce the low 32 bits of a multiply.
+// TIL: The results of a signed multiply and an unsigned multiply
+// differ only in the upper 32 bits; the low 32 bits remain the same for both.
+// So we can use the MUL and MLA ops unchanged for signed and unsigned operands.
+static void exec_MUL(uint32_t instruction)
+{
+    if (condition_check(curr_state, get_bits(instruction, 31, 28))) {
```

Comment:
remove condition_checks from all your instructions, we're doing these on decoder stage now